reactor-core/src/test/java/reactor/core/publisher/MonoCacheInvalidateWhenTest.java

```diff
@@ -263,7 +264,8 @@ void concurrentTriggerAndDownstreamSubscriber() {
 			if (sub2Seen == 1) lessInteresting++;
 		}
 		//smoke assertion: we caught more than 50% of interesting cases
-		assertThat(lessInteresting)
+		//however there is no guarantee we can't see more
+		assumeThat(lessInteresting)
```

Comment:
`assumeThat` is to skip a test under a condition that is evaluated at runtime... I wouldn't use that here. maybe a log if the number of "lessInteresting" cases goes beyond `loops / 2` instead of an assertion?

it sounds like the goal / expectation was to trigger at least 50% of interesting cases ? maybe we can just check that `lessInteresting` `isBetween(0, loops - 1)` instead? (at least 1 interesting case asserted).

Reply:
`isBetween` checks that number of events is between 0 and 50%. In reality - that number can be 0 as well as 100% since verification relies on random event but with finit number of observations.

If we put that on clear statistic - both events have equal probability. However, that is true for infinit number of experiments. If we have fixed number of experiments, we have 0 guarantees to get exactly 50 as well no guarantee we always see less or more.

Thus, this test from statistical verification is invalid and gives nothing.

Reply:
ok. I think as long as the `assumeThat` is the last statement the test should report any meaningful failure. it will just appear as skipped if it found no failure but couldn't make sure that interesting state was reached and asserted half the time